File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        int max = 0;
        int leftPos = -1, rightPos = -1, currMax = 0;
        
        for(int i = 0; i < seats.length; i++){
            if(seats[i] == 1){
                leftPos = -1;
                rightPos = -1;
            }
            else{
                if(leftPos == -1){
                    leftPos = i;
                    rightPos = i;
                }
                else{
                    rightPos = i;
                }
                
                currMax = rightPos-leftPos+1;
                //updating max based on condition
                if(leftPos == 0 || rightPos == seats.length-1){
                    if(currMax > max)
                        max = currMax;
                }
                else{
                    int mid = leftPos + (rightPos-leftPos)/2;
                    if(mid-leftPos+1 > max)
                        max = mid-leftPos+1;
                }
            }
        }
        
       return max;
    }
}

File: 849-maximize-distance-to-closest-person/849-maximize-distance-to-closest-person.cs
public class Solution {
    public int MaxDistToClosest(int[] seats) {
        
        int max = 0;
        
        int cl = -1, cr = -1, cmax = 0;
        
        for(int i = 0; i < seats.Length; i++){
            if(seats[i] == 1){
                cl = -1;
                cr = -1;
            }
            else{
                if(cl == -1){
                    cl = i;
                    cr = i;
                }
                else{
                    cr = i;
                }
                
                cmax = cr-cl+1;
                //updating max based on condition
                if(cl == 0 || cr == seats.Length-1){
                    if(cmax > max)
                        max = cmax;
                }
                else{
                    int mid = cl + (cr-cl)/2;
                    if(mid-cl+1 > max)
                        max = mid-cl+1;
                }
            }
        }
        
       return max;
        
    }
}